File: snow/biz_rule.py
import requests
import json

url = "https://api.github.com/repos/BAC/test-repo/dispatches"
access_token = "<GitHub Personal Access Token>"

payload = {
    "event_type": "my_event_type",
    "client_payload": {
        "appID": "app",
        "region": "eus"
    }
}

headers = {
    "Accept": "application/vnd.github.v3+json",
    "Authorization": f"Bearer {access_token}"
}

response = requests.post(url, json=payload, headers=headers)

if response.status_code == 204:
    print("Repository dispatch event triggered successfully")
else:
    print(f"Failed to trigger repository dispatch event. Status code: {response.status_code}")

File: snow/biz_rule.js
(function executeRule(current, previous /*null when async*/) {

    var url = "https://api.github.com/repos/BAC/test-repo/dispatches";
    var accessToken = "<GitHub Personal Access Token>";
  
    var payload = {
      event_type: "my_event_type",
      client_payload: {
        appID: "app",
        region: "eus"
      }
    };
  
    var headers = {
      'Content-Type': 'application/json',
      'Authorization': 'Bearer ' + accessToken
    };
  
    var response = new sn_ws.RESTMessageV2();
    response.setEndpoint(url);
    response.setHttpMethod("POST");
    response.setRequestHeader("Accept", "application/vnd.github.v3+json");
    response.setRequestBody(JSON.stringify(payload));
    response.setRequestHeader("Authorization", "Bearer " + accessToken);
  
    var httpResponse = response.execute();
    var responseBody = httpResponse.getBody();
    
    gs.info("Response Body: " + responseBody);
    
  })(current, previous);
  